kafka-ui-react-app/src/components/Schemas/New/New.tsx

```diff
@@ -33,14 +35,20 @@ const New: React.FC<NewProps> = ({ createSchema }) => {
           schema,
           schemaType,
         });
-        history.push(clusterSchemaPath(clusterName, subject));
       } catch (e) {
         // Show Error
       }
     },
     [clusterName]
   );
 
+  React.useEffect(() => {
+    if (isSubmitting && isSchemaCreated) {
+      const { subject } = getValues();
+      history.push(clusterSchemaPath(clusterName, subject));
+    }
+  }, [isSubmitting, isSchemaCreated, clusterName]);
+
```

Comment:
Previous implementation should work if you throw an error in the end of catch block of `createSchema` thunk. Just like you do for `updateSchema`.